s2a/src/test/resources/config.cnf

```diff
@@ -0,0 +1,17 @@
+[req]
+distinguished_name = req_distinguished_name
+req_extensions = req_ext
+
+[req_distinguished_name]
+countryName                     = Country Name (2 letter code)
+stateOrProvinceName             = State or Province Name (full name)
+localityName                    = Locality Name (eg, city)
+organizationalUnitName          = Organizational Unit Name (eg, section)
+commonName                      = Common Name (eg, your name or your server\'s hostname)
+emailAddress                    = Email Address
+
+[req_ext]
+subjectAltName = @alt_names
+
+[alt_names]
+IP.1 = ::
```

Comment:
What is happening here? How would this ever work?

Reply:
This is so that the generated certs have localhost IP addr, so that certificate verification passes in tests

Reply:
Isn't localhost `::1`?

Reply:
This is true. I ran a local test modifying the IP to be `::1` and regenerating these certs, confirming our tests still pass. I'll modify this in a separate PR along with the other comments on the EC cert data: https://github.com/grpc/grpc-java/pull/11540#discussion_r1778823717 .

I'll also update our go client tests for consistency: https://github.com/google/s2a-go/blob/main/testdata/config.cnf

Reply:
Done in https://github.com/grpc/grpc-java/pull/11583/commits/7a879a2d3c0b97f183a8ec0bae181f7136256bc7

Reply:
I think that's the wrong conclusion. It means IP matching wasn't actually used. So if it should be and wasn't, then you need to find that hostname matching bug and then we'll also fix this cert. If it was matching already, then the IP wasn't used.

Reply:
FWIW, we use `managedChannelBuilder.overrideAuthority()` to choose what we expect in the cert in tests. It lets us use one name for address resolution and another for TLS.

Reply:
Sorry for the confusion and thanks for pointing out. To be clear, out tests rely on the fact that Common Name is set to localhost (not actually in this config, but when this config is used to create the CSR, it is set. I noted this in the README). So that's why this field was left as ::, and had no effect on tests when set to ::1.

Reply:
Then let me repeat my initial question: What is happening here? What is it doing?

Reply:
We use the config to generate certs to be used for client & S2A when testing mTLS to S2A. This is used in an [integration test](https://github.com/grpc/grpc-java/blob/master/s2a/src/test/java/io/grpc/s2a/internal/handshaker/IntegrationTest.java#L145) (also in S2AChannelCredentialsTest unit test, but that's just a unit test, no channel actually gets established between the client and S2A). As part of mTLS peer cert verification process, it is required that certificate CommonName matches the hostname (which is localhost).

Reply:
I don't understand. This is alt_names aka SAN, which is not CommonName. And `::` is not `::1` and that would not validate if the client used "localhost". So that all seems irrelevant.

Reply:
(As I said before, as a hint for integration tests, don't put localhost in the cert, which is actually invalid. Instead, use `managedChannelBuilder.overrideAuthority()`. It is available in all languages; [`WithAuthority()` in Go](https://pkg.go.dev/google.golang.org/grpc#WithAuthority).)